### PR TITLE
Fixed #28132 -- Made MultiPartParser ignore filenames with trailing slash.

### PR DESCRIPTION
**Pull request by mbrown1413**
Trac ticket [#28132](https://code.djangoproject.com/ticket/28132).

### CODE REVIEW

File: django/http/multipartparser.py [hunk 1]
```diff
@@ -292,7 +294,7 @@ def handle_file_complete(self, old_field_name, counters):
         """
         for i, handler in enumerate(self._upload_handlers):
             file_obj = handler.file_complete(counters[i])
-            if file_obj:
+            if file_obj is not None:
```
Comment (felixxm) [inline django/http/multipartparser.py:297]: This is a separate issue, can we move it to a separate PR/commit? Also, test is required.

Comment (felixxm) [reply django/http/multipartparser.py:297]: Yes I think we need a separate ticket. I'm not sure if there is a valid use for custom handlers that return files with an empty filenames :thinking: I removed the 2nd commit.

**Comment by felixxm** [comment]
@mbrown1413 Thanks :+1: Welcome aboard :boat:

**Comment by smithdc1** [comment]
Hi @felixxm 
This seems to be very similar to [Ticket #18150](https://code.djangoproject.com/ticket/18150) 🤔.

I've not really delved into this area before but adding `backslash.jpg\\` to the test list this PR amends seems to work?

**Comment by felixxm** [comment]
@smithdc1 https://code.djangoproject.com/ticket/18150#comment:10
